src/maxdiffusion/max_utils.py

```diff
@@ -206,6 +206,22 @@ def upload_blob(destination_gcs_name, source_file_name):
   blob = bucket.blob(prefix_name)
   blob.upload_from_filename(source_file_name)
 
+def walk_and_upload_blobs(config, output_dir):
+  user_dir = os.path.expanduser('~')
+  uploaded_files = set()
+  for root, _, files in os.walk(os.path.abspath(output_dir)):
+    for file in files:
+      file_to_upload = os.path.join(root, file)
+      if file_to_upload in uploaded_files:
+        continue
```

Comment:
Are there files that may need to be overwritten? what's the use case when file is already there?

Reply:
This is because how `for root, _, files in os.walk(os.path.abspath(output_dir)):` and the concatenation that follows works, it creates duplicates of the file_to_upload, ending up uploading the same file multiple times and this prevents it.

Reply:
thanks for the explanation. LGTM for now, but maybe we can improve this later.